internal/batch/batch_manager.go

```diff
@@ -132,280 +133,228 @@ func (bm *batchManager) NewMessages() chan<- int64 {
 	return bm.newMessages
 }
 
-func (bm *batchManager) restoreOffset() (err error) {
-	var offset *fftypes.Offset
-	for offset == nil {
-		offset, err = bm.database.GetOffset(bm.ctx, fftypes.OffsetTypeBatch, msgBatchOffsetName)
-		if err != nil {
-			return err
-		}
-		if offset == nil {
-			_ = bm.database.UpsertOffset(bm.ctx, &fftypes.Offset{
-				Type:    fftypes.OffsetTypeBatch,
-				Name:    msgBatchOffsetName,
-				Current: 0,
-			}, false)
-		}
-	}
-	bm.offsetID = offset.RowID
-	bm.offset = offset.Current
-	log.L(bm.ctx).Infof("Batch manager restored offset %d", bm.offset)
-	return nil
-}
+func (bm *batchManager) getProcessor(txType fftypes.TransactionType, msgType fftypes.MessageType, group *fftypes.Bytes32, namespace string, identity *fftypes.Identity) (*batchProcessor, error) {
+	bm.dispatcherMux.Lock()
+	defer bm.dispatcherMux.Unlock()
 
-func (bm *batchManager) removeProcessor(dispatcher *dispatcher, key string) {
-	dispatcher.mux.Lock()
-	delete(dispatcher.processors, key)
-	dispatcher.mux.Unlock()
-}
-
-func (bm *batchManager) getProcessor(batchType fftypes.MessageType, group *fftypes.Bytes32, namespace string, identity *fftypes.Identity) (*batchProcessor, error) {
-	dispatcher, ok := bm.dispatchers[batchType]
+	dispatcherKey := bm.getDispatcherKey(txType, msgType)
+	dispatcher, ok := bm.dispatchers[dispatcherKey]
 	if !ok {
-		return nil, i18n.NewError(bm.ctx, i18n.MsgUnregisteredBatchType, batchType)
+		return nil, i18n.NewError(bm.ctx, i18n.MsgUnregisteredBatchType, dispatcherKey)
 	}
-	dispatcher.mux.Lock()
-	key := fmt.Sprintf("%s:%s:%s[group=%v]", namespace, identity.Author, identity.Key, group)
-	processor, ok := dispatcher.processors[key]
+	name := fmt.Sprintf("%s|%s|%v", namespace, identity.Author, group)
```

Comment:
Small thing, but maybe there should be a `getProcessorKey` next to `getDispatcherKey`.